### PR TITLE
Updated entity resolution

### LINKED ISSUE #978: Intra-Flight Entity Resolution
using reservation system provided unique IDs to resolve passengers within a "family" of PNR messages.

### PR DESCRIPTION
**Pull request by kasimhelil**
Resolves #978

### CODE REVIEW

File: gtas-parent/gtas-commons/src/main/java/gov/gtas/repository/FlightPassengerRepository.java [hunk 1]
```diff
@@ -23,4 +25,81 @@
...
+             "LEFT JOIN FETCH pax.seatAssignments " +
+             "LEFT JOIN FETCH pax.bags " +
+             "LEFT JOIN FETCH pax.flightPaxList " +
+             "LEFT JOIN FETCH pax.tickets " +
+             "LEFT JOIN FETCH pax.bookingDetails " +
+             "LEFT JOIN FETCH pax.passengerDetails " +
+             "LEFT JOIN FETCH pax.passengerTripDetails " +
+             "WHERE (UPPER(fp.passenger.passengerDetails.firstName) = UPPER(:firstName) " +
+             "AND UPPER(fp.passenger.passengerDetails.lastName) = UPPER(:lastName) " +
+             "AND fp.passenger.passengerDetails.dob = :dob) " +
```
Comment (originalname51) [inline gtas-parent/gtas-commons/src/main/java/gov/gtas/repository/FlightPassengerRepository.java]: fp.passenger -> pax

File: gtas-parent/gtas-commons/src/main/java/gov/gtas/repository/FlightPassengerRepository.java [hunk 2]
```diff
@@ -23,4 +25,81 @@
...
+             "JOIN FETCH fp.passenger pax " +
+             "LEFT JOIN FETCH pax.documents " +
+             "LEFT JOIN FETCH pax.seatAssignments " +
+             "LEFT JOIN FETCH pax.bags " +
+             "LEFT JOIN FETCH pax.flightPaxList " +
+             "LEFT JOIN FETCH pax.tickets " +
+             "LEFT JOIN FETCH pax.bookingDetails " +
+             "LEFT JOIN FETCH pax.passengerDetails " +
+             "LEFT JOIN FETCH pax.passengerTripDetails " +
+             "WHERE (UPPER(fp.passenger.passengerDetails.firstName) = UPPER(:firstName) " +
```
Comment (originalname51) [inline gtas-parent/gtas-commons/src/main/java/gov/gtas/repository/FlightPassengerRepository.java]: fp.passenger

File: gtas-parent/gtas-loader/src/main/java/gov/gtas/services/ServiceUtil.java [hunk 3]
```diff
@@ -13,56 +13,138 @@
...
 
-    @Autowired
-    PassengerRepository passengerRepository;
+	}
 
-    @Autowired
-    FlightPassengerRepository flightPassengerRepository;
+	@Override
+	@Transactional()
+	public Passenger findPassengerOnFlight(Flight f, PassengerVo pvo) {
```
Comment (originalname51) [inline gtas-parent/gtas-loader/src/main/java/gov/gtas/services/ServiceUtil.java]: The if statements here are complicated and more readable when resolved to boolean values.

File: gtas-parent/gtas-loader/src/main/java/gov/gtas/services/ServiceUtil.java [hunk 4]
```diff
@@ -13,56 +13,138 @@
...
+		String gender = pvo.getGender();
+		String recordLocator = pvo.getPnrRecordLocator();
+		String ref = pvo.getPnrReservationReferenceNumber();
+		boolean found = false;
 
-    public String getCountry(String airport) {
+		//Resolve passenger by Flight -> pnr recordLocator# -> pnr REF number       
+		if (recordLocator != null && ref != null) {
+			pax = flightPassengerRepository.getPassengerUsingREF(flightId, ref, recordLocator);	
+			found = (pax != null && pax.size() > 0);
```
Comment (originalname51) [inline gtas-parent/gtas-loader/src/main/java/gov/gtas/services/ServiceUtil.java]: !CollectionUtils.isEmpty(pax) says the same thing.

File: gtas-parent/gtas-loader/src/test/java/gov/gtas/services/ServiceUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+		FlightPassenger flightPassenger = new FlightPassenger();
+		flightPassenger.setPassenger(passenger);
+
+		List<FlightPassenger> list = new ArrayList<FlightPassenger>();
+		list.add(flightPassenger);
+
+		return list;
+	}
+
+	private String createPassengerIdTag(PassengerVo pvo) {
```
Comment (originalname51) [inline gtas-parent/gtas-loader/src/test/java/gov/gtas/services/ServiceUtilTest.java]: Is this the same as line 70?

Comment (originalname51) [reply gtas-parent/gtas-loader/src/test/java/gov/gtas/services/ServiceUtilTest.java]: Whoops, wrong file!

Comment (originalname51) [reply gtas-parent/gtas-loader/src/test/java/gov/gtas/services/ServiceUtilTest.java]: This method should call the other passenger id tag in production code.

Comment (kasimhelil) [reply gtas-parent/gtas-loader/src/test/java/gov/gtas/services/ServiceUtilTest.java]: `createPassengerIdTag` is a helper method in **ServiceUtil** class, and I believe the convention for the scope of a helper method should be the class that defines the method.

**Pull request merged**